db/migrate/20240312122903_make_stripe_payment_intent_generic.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+class MakeStripePaymentIntentGeneric < ActiveRecord::Migration[7.1]
+  def change
+    change_table :stripe_payment_intents do |t|
+      t.rename :user_id, :initiated_by_id # Make the purpose of the user_id column more clear
+      t.rename :confirmed_by_id, :confirmation_source_id
+      t.rename :confirmed_by_type, :confirmation_source_type
+      t.rename :canceled_by_id, :cancellation_source_id
+      t.rename :canceled_by_type, :cancellation_source_type
+
+      t.string :wca_status
+      t.string :payment_record_type
+      t.integer :payment_record_id
+    end
+
+    rename_table :stripe_payment_intents, :payment_intents
+    rename_column :stripe_records, :status, :stripe_status
+    rename_column :stripe_records, :api_type, :record_type
+    rename_column :paypal_records, :status, :paypal_status
+
+    reversible do |direction|
+      direction.up do
+        RegistrationPayment.update_all(receipt_type: 'StripeRecord')
+
+        PaymentIntent.update_all(payment_record_type: 'StripeRecord')
```

Comment:
Haven't we already done this part in https://github.com/thewca/worldcubeassociation.org/pull/9067 ?!

Reply:
Yes we absolutely have. Not sure how I ended up with records that hadn't been affected by it - change reverted.